docs/libvirt-pod-networking.md

```diff
@@ -0,0 +1,118 @@
+# Using CNI Interfaces and libvirt for networking
+
+Author: Fabian Deutsch \<fabiand@redhat.com\>
+
+## Introduction
+
+Today networking in KubeVirt is not integrated with Kubernetes, a user can
+rather choose to create a specific VM Spec definition to create a particular
+libvirt domxml, to reuse an existing physical device.
+But this is obviously suboptimal
+
+- It is not integrated with Kubernetes
+- It does not provide connectivity with pods
+- It requires to specify host specific bits (the network device to use)
+
+The purpose of this document is to describe one approach of how we can provide
+a minimal integrated solution, with few workarounds.
+
+**Note:** This approach is rough, and not optimized, but the rule of thumb was
+to find a robust solution which works with Kubernetes today.
+
+
+## Use-case
+
+The primary use-case is to allow VM to get connected to the pod network.
+
+
+## API
+
+The API is pretty clear: As usual an interface will be defined.
+To indicate that Pod interfaces shall be used for networking, a new type
+`pod-network` is introduced.
+All other fields, concerning the properties of the virtual NIC, can be used
+as usual.
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: pod-network
+        mac:
+          address: 00:11:22:33:44:55
+        model:
+          type: virtio
+```
+
+
+## Implementation
+
+Prerequisites
+- libvirt is using Pod networking
+- A CNI proxy to call the hosts CNI is available (this can be part of this
+  designs implementation)
+
+Conceptually the implementation works by assuming that every vNIC will be
+mapped to a _new_ Pod interface. To achieve this, for every vNIC a _new_ Pod
+interface is requested through CNI.
+Any after creation of the Pod created interfaces will from now on be called
+_VM interface_ to differentiate them from the the original during pod creation
+allocated Pod interface.
```

Comment:
During this document we call each of these _new_ Pod interfaces  _VM interface_, in order to differentiate them from the the originally-allocated Pod interface (`eth0`). The original Pod interface (`eth0`) is never modified by Kubevirt, and can be used to access libvirtd (through the libvirtd Pod IP) or to provide VM-centric services through the VM Pod IP.